docs/ocis/adr/0011-global-url-format.md

```diff
@@ -0,0 +1,210 @@
+
+Space Registry resolution can happen at the client side (i.e: the client keeps a list of space name -> space id [where space id = storageid + nodeid]; the client queries a SR) or server side. Server side is more resilient due to clients can have limited networking; for instance if they are running on a tight intranet.
+
+### Mixed Global URLs
+
+While ID based space URLs can be made more readable by shortening the IDs they only start to reveal context when an alias is used instead of the space id. These aliases however have to be unique identifiers. These aliases shouly live in namespaces like `/workspaces/marketing` and `/personal/marketing` to make phishing attacks harder (in this case a user that registered with the username `marketing`). But namespaced aliases is semantically equivalent to ... a path hierarchy.
+
+When every space has a namespaced alias and a relative path we can build a global namespace:
+
+| URL | comment |
+|-|-|
+| `https://<host>/files</namespaced/alias></relative/path/to/resource>?id=<resource_id>` | the pattern, `/files` might become optional |
+| `https://demo.owncloud.com/files/personal/einstein/?id=b78c2044-5b51-446f-82f6-907a664d089c:194b4a97-597c-4461-ab56-afd4f5a21607` | root of user `einstein` |
+| `https://demo.owncloud.com/files/personal/einstein/relative/path/to/resource?id=b78c2044-5b51-446f-82f6-907a664d089c:194b4a97-597c-4461-ab56-afd4f5a21608` | sub folder `/relative/path/to/resource` |
+| `https://demo.owncloud.com/files/public/kcZVYaXr7oZ66bg/relative/path/to/resource` | sub folder `/relative/path/to/resource` in public link with token `kcZVYaXr7oZ66bg` |
+| `https://demo.owncloud.com/files/public/kcZVYaXr7oZ66bg/relative/path/to/resource` | sub folder `/relative/path/to/resource` in public link with token `kcZVYaXr7oZ66bg` |
+
+`</namespaced/alias></relative/path/to/resource>` is the global path in the CS3 api. The CS3 Storage Registry is responsible by managing the mount points.
+
+In order to be able to copy and paste URLs all resources must be uniquely identifyable:
+
+* Instead of `/home` the URL always has to reflect the user: `/personal/einstein`
+* Public links can use `/public/<token>`
+* workspaces can use `/workspaces/<alias>` or `/workspaces/<additional>/<classification>/<alias>` where the hierarchy is given by the organization
+* experiments can use `/experiments/<alias>`
+* research institutes could set up `/papers/<researchgroup>/<alias>`
+* trash could be accessed by prefixing the namespace alias with `/trash`? or using `/trash/<space_id>`
+* instead of a namespaced alias a storage space id could be used with a generic `/space/<space_id>` namespace
+
+The alias namespace hierarchy and depth can be pre determined by the admin. Even if aliases change the `id` parameter prevents bookmarks from breaking. A user can decide to build a different hierarchy by using his own registry. 
```

Comment:
> A user can decide to build a different hierarchy by using his own registry.

Who is the user here? an Admin?

Reply:
an admin or an end user that has access to multiple registries and wents to mix and mount storages at his discretion. I'm not saying he has to do that or that it is a good feeture. My argument is that for a truly federated storage the storage registry can live anywhere, it can contain storage spaces from everywhere, and the user is in complete control how he sets up his namespace. An admin might, and should organize the storage spaces in his instance as best fits the organization. While he is the admin of the registry, the clients ultimately have to decide what to sync where.